src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -447,14 +603,10 @@ public override bool MoveNext()
                         state = 2;
                         goto case 2;
                     case 2:
-                        while (_enumerator.MoveNext())
+                        if (_enumerator.MoveNext())
```

Comment:
That would change the behaviour if the source list is altered during enumeration. (I've never been convinced the dangers of that are such as to require an outright ban, but it's way past the point of debating that). The test `Select_SourceListGetsModifiedDuringIteration_ExceptionIsPropagated` checks on this.